RavenDB-12823 should mutate the revision flags to include `Conflicted…

karmeli87 opened the pull request:
…` once conflict is resolved

karmeli87 committed:
RavenDB-12823 should mutate the revision flags to include `Conflicted` once conflict is resolved

ayende commented on src/Raven.Server/Documents/Revisions/RevisionsStorage.cs:
```diff
@@ -721,10 +723,13 @@ private static void DeleteCountOfRevisions(DocumentsOperationContext context, Sl
             {
-                var revisionExists = table.VerifyKeyExists(changeVectorSlice);
+                var revisionExists = table.ReadByKey(changeVectorSlice,out var tvr);
                 // Revisions are immutable.
```
wrong comment?

karmeli87 replied:
yes, removed

ayende commented on src/Raven.Server/Documents/Revisions/RevisionsStorage.cs:
```diff
@@ -769,6 +774,52 @@ private static void DeleteCountOfRevisions(DocumentsOperationContext context, Sl
+        private void MarkRevisionsAsConflictedIfNeeded(DocumentsOperationContext context, Slice lowerId, Slice idSlice, DocumentFlags flags, TableValueReader tvr, Table table,
+            Slice changeVectorSlice)
+        {
+            // Revisions are immutable, but if there was a conflict we need to update the flags accordingly with the `Conflicted` flag.
```
Why is that the case? The revision itself isn't conflicted, after all. 
Or is it that it was conflicted on another node?

karmeli87 replied:
This is relevant in the case when revisions are enabled.
So node A generates a document with a revision, and node B also.
When the conflict is resolved, we need to add the `Conflicted` flag to an already existing revision.

karmeli87 committed:
RavenDB-12823 add more tests

karmeli87 committed:
RavenDB-12823 the resolved document should preserve the modified time

otherwise we could have a race between two nodes that resolving the same conflict to a different result.
